crates/net/network/src/network.rs

```diff
@@ -344,6 +328,32 @@ impl NetworkSyncUpdater for NetworkHandle {
     }
 }
 
+impl EthNetwork for NetworkHandle {
+    fn set_tx_handle(&self, handle: TransactionsHandle) {
+        let mut tx_handle = self.inner.tx_handle.lock();
+        *tx_handle = Some(handle);
+    }
+
+    fn tx_handle(&self) -> Option<TransactionsHandle> {
+        self.inner.tx_handle.lock().clone()
```

Comment:
should I return a reference here

Reply:
we can replace the mutex if we add another message to
https://github.com/paradigmxyz/reth/blob/be6d5057451501285f9cfa95682bfdcbd055be3d/crates/net/network/src/transactions/mod.rs#L1538-L1538

with a oneshot::Sender<TransactionsHandle> 

and the same to https://github.com/paradigmxyz/reth/blob/be6d5057451501285f9cfa95682bfdcbd055be3d/crates/net/network/src/network.rs#L396-L396

so that the NetworkManager will essentially delegate this